### PR TITLE
Command Palette: Maintain page state when changing theme

### LINKED ISSUE #56840: Command Palette: Explore state is reset after theme changing
**Steps to reproduce**:

* Open Explore and run a query
* Open Command Palette <kbd>cmd</kbd> + <kbd>K</kbd>
* Switch the theme

Result: The theme updates but the page is refreshed and the query editor state is lost
Expected: The theme updates and the state of the page doesn't change

https://user-images.githubusercontent.com/745532/195527015-a516770b-f41e-4b29-8963-2cd1eb3fde2c.mov

**Technical info**:

Command Palette [refreshes the page](https://github.com/grafana/grafana/blob/v9.2.0/public/app/features/commandPalette/actions/global.static.actions.ts#L43) when the theme changes. It affects not only Explore but any page that loses state upon refresh. We should probably call [changeTheme](https://github.com/grafana/grafana/blob/v9.2.0/public/app/core/services/toggleTheme.ts) instead if it’s feasible. 

**Environment**:
- Grafana version: 9.2.0

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- stops doing a full page reload when changing the theme

**Why do we need this feature?**

- allows us to maintain state in e.g. Explore

**Who is this feature for?**

- everyone! 🙌 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/56840

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/services/toggleTheme.ts [hunk 1]
```diff
@@ -7,11 +7,10 @@ import { contextSrv } from '../core';
 
 import { PreferencesService } from './PreferencesService';
 
-export async function toggleTheme(runtimeOnly: boolean) {
```
Comment (gelicia) [inline public/app/core/services/toggleTheme.ts:10]: Would we want to rename this file too?

**Review by gelicia — APPROVED** [review]
Looks awesome! It looks like this approach avoids the bugs you see when toggling themes with keybindings (`c t`, `c r`)as well.

**Comment by ashharrison90** [comment]
@gelicia it uses exactly the same function as the `c t` keybindings now (so should have the same bugs 😅 )... but we did just recently fix some bugs with `c t` [here](https://github.com/grafana/grafana/pull/59031) so maybe that's why they're fixed?

**Comment by gelicia** [comment]
> @gelicia it uses exactly the same function as the `c t` keybindings now (so should have the same bugs 😅 )... but we did just recently fix some bugs with `c t` [here](https://github.com/grafana/grafana/pull/59031) so maybe that's why they're fixed?

Hmm yeah it is using the same code but I still see the partial theme on the home screen using the keybindings and not the command palette.
Screen Shot 2022-12-05 at 8 07 50 AM

**Comment by ashharrison90** [comment]
@gelicia is that using `c r` instead of `c t`? 🤔 

command palette and `c t` _should_ behave exactly the same. `c r` is a little weird because it doesn't make any backend call, so you end up with this half state... that's part of the reason why we haven't documented it anywhere as well 😬

**Comment by gelicia** [comment]
> `c r` is a little weird because it doesn't make any backend call, so you end up with this half state... that's part of the reason why we haven't documented it anywhere as well 😬

Ah got it, didn't realize there was a distinction between the keybindings. Thanks!

**Pull request merged**